content/tutorials/core-ledger/create-custom-transactors.md

```diff
@@ -0,0 +1,389 @@
+
+Transactors follow a basic order of operations:
+
+1. Access a _view_ into a serialized type ledger entry (SLE).
+2. Update, erase, or insert values in the _view_.
+3. Apply the finalized changes from the _view_ to the ledger.
+
+**Note:** _Views_ are sandboxes into ledgers. Transactors make all necessary error checks and changes in sandboxes, not directly on the ledger. After values are finalized, changes are applied atomically to the ledger.
+
+This tutorial uses the existing `CreateCheck` transactor as an example. You can view the source files here:
+
+- [Header File](https://github.com/XRPLF/rippled/blob/master/src/ripple/app/tx/impl/CreateCheck.h)
+- [CPP File](https://github.com/XRPLF/rippled/blob/master/src/ripple/app/tx/impl/CreateCheck.cpp)
+
+
+## Header File
+
+Create a header file in this format:
+
+```c++
+namespace ripple {
+
+class CreateCheck : public Transactor
+{
+public:
+    static constexpr ConsequencesFactoryType ConsequencesFactory{Normal};
+
+    explicit CreateCheck(ApplyContext& ctx) : Transactor(ctx)
+    {
+    }
+
+    static NotTEC
+    preflight(PreflightContext const& ctx);
+
+    static TER
+    preclaim(PreclaimContext const& ctx);
+
+    TER
+    doApply() override;
+};
+
+}  // namespace ripple
+```
+
+Initializing the transactor with `ApplyContext` gives it access to:
+
+- The transaction that triggered the transactor.
+- A view of the SLE.
+- A journal to log errors.
+
+
+## CPP File
+
+### 1. Add a `preflight` function.
+
+The `preflight` function checks for errors before accessing the ledger. It should reject invalid and incorrectly formed transactions.
+
+- `PreflightContext` doesn't have a view of the ledger.
+- Use bracket notation to retrieve fields from ledgers and transactions:
+        
+        auto const curExpiration = (*sle*)[~sfExpiration];
+        (*sle)[sfBalance] = (*sle)[sfBalance] + reqDelta;
+            
+    **Note:** The `~` symbol returns an optional type.
+
+- You can view ledger and transaction schemas here:
+    - [`LedgerFormats.cpp`](https://github.com/XRPLF/rippled/blob/master/src/ripple/protocol/impl/LedgerFormats.cpp)
+    - [`TxFormats.cpp`](https://github.com/XRPLF/rippled/blob/master/src/ripple/protocol/impl/TxFormats.cpp)
+
+- `rippled` summarizes transaction results with result codes. See: [Transaction Results](transaction-results.html)
+
+```c++
+CreateCheck::preflight(PreflightContext const& ctx)
+{
+    // Check if this amendment functionality is enabled on the network.
+    if (!ctx.rules.enabled(featureChecks))
+        return temDISABLED;
+
+    NotTEC const ret{preflight1(ctx)};
+    if (!isTesSuccess(ret))
+        return ret;
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+    {
+        // There are no flags (other than universal) for CreateCheck yet.
+        JLOG(ctx.j.warn()) << "Malformed transaction: Invalid flags set.";
```

Comment:
Perhaps there should be a note about logging here - to explain that this shows up in the debug log.